nilearn/datasets/atlas.py

```diff
@@ -320,104 +325,274 @@ def fetch_atlas_harvard_oxford(atlas_name, data_dir=None,
-
-    atlas_img = check_niimg(atlas_img)
-    if lateralized:
-        return Bunch(maps=atlas_img, labels=names)
-
-    atlas = get_data(atlas_img)
-
-    labels = np.unique(atlas)
+        new_idx = int(label.get('index')) + 1
+        if new_idx in names:
+            raise IndexError(
+                f"Duplicate index {new_idx} for labels "
+                f"'{names[new_idx]}', and '{label.text}'")
+        names[new_idx] = label.text
+    names = [item[1] for item in sorted(names.items())]
+    return atlas_img, names, is_lateralized
+
+
+def _merge_probabilistic_maps_juelich(atlas_data, names):
+    """Helper function for fetch_atlas_juelich.
+    This function handles probabilistic juelich atlases
+    when symmetric_split=False. In this situation, we need
+    to merge labels and maps corresponding to left and right
+    regions.
+    """
+    new_names = np.unique([re.sub(r" (L|R)$", "", name) for name in names])
+    new_name_to_idx = {k: v - 1 for v, k in enumerate(new_names)}
+    new_atlas_data = np.zeros((*atlas_data.shape[:3],
+                               len(new_names) - 1))
+    for i, name in enumerate(names):
+        if name != "Background":
+            new_name = re.sub(r" (L|R)$", "", name)
+            new_atlas_data[..., new_name_to_idx[new_name]] += (
+                atlas_data[..., i - 1])
+    return new_atlas_data, new_names
+
+
+def _merge_labels_juelich(atlas_data, names):
+    """Helper function for fetch_atlas_juelich.
+    This function handles 3D atlases when symmetric_split=False.
+    In this case, we need to merge the labels corresponding to
+    left and right regions.
+    """
+    new_names = np.unique([re.sub(r" (L|R)$", "", name) for name in names])
+    new_names_dict = {k: v for v, k in enumerate(new_names)}
+    new_atlas_data = atlas_data.copy()
+    for label, name in enumerate(names):
+        new_name = re.sub(r" (L|R)$", "", name)
+        new_atlas_data[atlas_data == label] = new_names_dict[new_name]
+    return new_atlas_data, new_names
+
+
+def _compute_symmetric_split(source, atlas_niimg, names):
+    """Helper function for both fetch_atlas_juelich and
+    fetch_atlas_harvard_oxford.
+    This function handles 3D atlases when symmetric_split=True.
+    """
+    atlas_data = get_data(atlas_niimg)
+    labels = np.unique(atlas_data)
     # Build a mask of both halves of the brain
-    middle_ind = (atlas.shape[0] - 1) // 2
-    # Put zeros on the median plane
-    atlas[middle_ind, ...] = 0
+    middle_ind = (atlas_data.shape[0]) // 2
     # Split every zone crossing the median plane into two parts.
-    left_atlas = atlas.copy()
-    left_atlas[middle_ind:, ...] = 0
-    right_atlas = atlas.copy()
-    right_atlas[:middle_ind, ...] = 0
+    left_atlas = atlas_data.copy()
+    left_atlas[middle_ind:] = 0
+    right_atlas = atlas_data.copy()
+    right_atlas[:middle_ind] = 0
+
+    if source == "Juelich":
+        for idx in range(len(names)):
+            if names[idx].endswith('L'):
```

Comment:
this condition is not needed anymore

Reply:
I think it is needed in order to have the labels converted from "label R" / "label L" to "Right label" / "Left label".

Note: I changed this in ac66400 because labels were not consistent, some were starting with "Right | Left" while others were ending with ",right part |,left part".

Reply:
then you could simply use `re.sub(r"(.*) L$", r"Left \1", name)`, without a need for the if?

(sorry if this is outdated something weird seems to have happened with the comments in this PR)